pulsar-transaction/coordinator/src/main/java/org/apache/pulsar/transaction/coordinator/impl/TxnLogBufferedWriter.java

```diff
@@ -198,33 +209,99 @@ public void asyncAddData(T data, AddDataCallback callback, Object ctx){
                     AsyncAddArgs.newInstance(callback, ctx, System.currentTimeMillis(), byteBuf));
             return;
         }
-        singleThreadExecutorForWrite.execute(() -> internalAsyncAddData(data, callback, ctx));
+        singleThreadExecutorForWrite.execute(() -> {
+            try {
+                internalAsyncAddData(data, callback, ctx);
+            } catch (Exception e){
+                log.error("Internal async add data fail", e);
```

Comment:
> Does it make sense to log the error instead of calling callback.addFailed?

It simply prints all the exceptions that have occurred in asynchronous tasks and are used to locate problems.

As long as the request is in the memory queue, it will eventually succeed or fail, we do not need redundant executive `callback.addFailed`